client/src/components/DocsNav/index.jsx

```diff
@@ -1,14 +1,73 @@
+import { useEffect, useState, Fragment } from 'react';
 import { NavLink } from 'react-router-dom';
+import { useWindowSize } from '../../hooks/useWindowSize';
 
 // styles
 import './docsnav.scss';
 
 export const DocsNav = ({ navigation }) => {
+    const [isOpen, setIsOpen] = useState(false);
+    const _windowSize = useWindowSize();
+
+    const toggleNav = () => {
+        setIsOpen((prevState) => !prevState);
+    };
+
+    useEffect(() => {
```

Comment:
It does make sense to use it. But, I wouldn't say it's better to use `useLayoutEffect` instead of `useEffect` because [React docs](https://reactjs.org/docs/hooks-reference.html#uselayouteffect) warns us against using it, unless you've got a really good reason. However, there are some edge cases where you might've to use `useLayoutEffect()` which is covered in this [blog post](https://kentcdodds.com/blog/useeffect-vs-uselayouteffect) by Kent